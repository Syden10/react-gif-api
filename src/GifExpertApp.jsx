import { useState } from 'react';
import { AddCategory, GifGrid } from './components/';

export default function GifExpertApp() {
  const [categories, setCategories] = useState(['cats']);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  const clearCategories = () => {
    setCategories([]);
  };

  return (
    <>
      <h1>Gif App</h1>
      <div className='control-box'>
        <div className='search-box'>
          <AddCategory onNewCategory={onAddCategory} />
        </div>
        <div className='btn-box'>
          <button onClick={clearCategories}>Clear Gifs</button>
        </div>
      </div>
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
}
