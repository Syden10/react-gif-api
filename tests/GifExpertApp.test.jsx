import { render, screen } from '@testing-library/react';
import GifExpertApp from '../src/GifExpertApp';

describe('<GifExpertApp /> tests', () => {
  it('should render <GifExpertApp /> correctly', () => {
    render(<GifExpertApp />);
    expect(screen.getByRole('heading', { level: 1 })).toMatchSnapshot();
    screen.debug();
  });
  const mockSetCategories = jest.fn();
  let categories = ['Category1', 'Category2'];

  beforeEach(() => {
    mockSetCategories.mockClear();
    categories = ['Category1', 'Category2']; // Reset categories before each test
  });

  it('should add new category to the beginning of the categories array if not already present', () => {
    const newCategory = 'Category3';
    const onAddCategory = (newCategory) => {
      if (categories.includes(newCategory)) return;
      mockSetCategories([newCategory, ...categories]);
    };

    onAddCategory(newCategory);

    expect(mockSetCategories).toHaveBeenCalledWith([
      'Category3',
      'Category1',
      'Category2',
    ]);
  });

  it('should not call setCategories if newCategory is already present in categories', () => {
    const newCategory = 'Category1';
    const onAddCategory = (newCategory) => {
      if (categories.includes(newCategory)) return;
      mockSetCategories([newCategory, ...categories]);
    };

    onAddCategory(newCategory);

    expect(mockSetCategories).not.toHaveBeenCalled();
  });
});
