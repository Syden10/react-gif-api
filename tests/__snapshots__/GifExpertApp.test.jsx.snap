// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp /> tests should render <GifExpertApp /> correctly 1`] = `
<h1>
  Gif App
</h1>
`;
